routers/api/v1/repo/pull.go

```diff
@@ -726,7 +727,15 @@ func MergePullRequest(ctx *context.APIContext) {
 	//     "$ref": "#/responses/error"
 
 	form := web.GetForm(ctx).(*forms.MergePullRequestForm)
-	pr, err := models.GetPullRequestByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	dbCtx, committer, err := db.TxContext()
```

Comment:
I think a better way is to move the transaction to some method in services

Reply:
This PR already changes a lot .........

Reply:
I'll find a better way ...

Reply:
If no other better choice, I prefer to keep the code as it is, and refactor separately. This PR already changes a lot.